python_modules/dagster/dagster/core/execution/plan/state.py

```diff
@@ -29,6 +30,40 @@ def get_version_dict_from_list(
         return {data.step_output_handle: data.version for data in step_output_versions}
 
 
+@whitelist_for_serdes
+class ParentRun(
```

Comment:
plz help me find a better name

Reply:
I feel like it might be bad to have "Parent" in the name at all. Like known execution state having a field called parent_state (or similar) makes sense, but parent_state being of type ParentState is weird (I would expect a field named parent_foo_bar to be of type FooBar, not ParentFooBar).

I'd be more in favor of something along the lines of ProducedOutputState or KnownOutputs or KnownOutputState (none of these specific options feel that amazing though)

Reply:
> I would expect a field named parent_foo_bar to be of type FooBar, not ParentFooBar)

I find this reasoning compelling.

I am hesitant to restrict it just to "output" state since we may need to add other information. It will almost certainly be different than the state we need to track for the current active run, so i dont want to just recurse `KnownExecutionState` (maybe should be called `ActiveExecutionState` or `KnownActiveExecutionState` but renaming serdes shit is a pain).

`PastExecutionInfo` 
`PastExecutionState`

Reply:
> I would expect a field named parent_foo_bar to be of type FooBar, not ParentFooBar)
good callout

`PastExecution*` sounds good to me as well